engine-src/Game/LambdaHack/Client/UI/HandleHumanLocalM.hs

```diff
@@ -681,20 +682,20 @@ selectWithPointerHuman = do
 repeatHuman :: MonadClientUI m => Int -> m ()
 repeatHuman n = do
   macro <- getsSession smacroBuffer
-  let nmacro k | k == 1 = reverse . fromLeft [] $ macro
+  let nmacro k | k == 1 = reverse . unMacro . fromLeft mempty $ macro
                -- Don't repeat macro while recording one.
                | otherwise = concat . replicate k $ nmacro 1
   modifySession $ \sess -> 
-    sess {slastPlay = nmacro n ++ slastPlay sess}
+    sess {slastPlay = (InGameMacro $ nmacro n) <> slastPlay sess}
```

Comment:
hlint would suggest `InGameMacro (nmacro n) <> slastPlay sess` (and also in other places). I'm like this version, too, but we'd need to adjust hlint rules.